### PR TITLE
[PyTorch] Cache all java classes

### PR DESCRIPTION
**Pull request by stu1130**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

**Review by frankfliu — APPROVED** [review]
This is good to have. But not really necessary. It's OK to be slow in exception condition.

File: pytorch/pytorch-native/src/main/native/ai_djl_pytorch_jni_cache.cc [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+#include "ai_djl_pytorch_jni_cache.h"
+
+jclass NULL_PTR_EXCEPTION_CLASS;
+jclass ILLEGAL_STATE_EXCEPTION_CLASS;
```
Comment (lanking520) [inline pytorch/pytorch-native/src/main/native/ai_djl_pytorch_jni_cache.cc:17]: Did we use this anywhere?

**Review by lanking520 — APPROVED** [review]
I guess we may need to globally improving the error message handling by using different exception classes.

File: pytorch/pytorch-native/src/main/native/ai_djl_pytorch_jni_cache.cc (hunk 1 above)
Comment (stu1130) [reply pytorch/pytorch-native/src/main/native/ai_djl_pytorch_jni_cache.cc:17]: just added the ILLEGAL_STATE_EXCEPTION_CLASS

**Pull request merged**